camayoc/tests/qcs/cli/test_sources.py

```diff
@@ -0,0 +1,944 @@
+            'type': source_type,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    qpc_source_add = pexpect.spawn(
+        'qpc source add --name {} --cred {} --hosts {} --type {}'
+        .format(name, cred_name, hosts, source_type)
+    )
+    assert qpc_source_add.expect('Source "{}" was added'.format(name)) == 0
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == 0
+
+    qpc_source_edit = pexpect.spawn(
+        'qpc source edit --name {} --cred {}'
+        .format(invalid_name, utils.uuid4())
+    )
+    qpc_source_edit.logfile = BytesIO()
+    assert qpc_source_edit.expect(
+        'Source "{}" does not exist.'.format(invalid_name)) == 0
+    assert qpc_source_edit.expect(pexpect.EOF) == 0
+    qpc_source_edit.close()
+    assert qpc_source_edit.exitstatus == 1
+
+
+@pytest.mark.parametrize('new_hosts', NEW_HOSTS)
+def test_edit_hosts(
+        isolated_filesystem, qpc_server_config, new_hosts, source_type):
+    """Edit a source's hosts.
+
+    :id: bfc1d4a8-e4af-406f-a70e-46674fa2c1d0
+    :description: Edit the hosts of a source entry.
+    :steps: Run ``qpc source edit --name <name> --hosts <newhosts>``
+    :expectedresults: The source's hosts must be updated.
+    """
+    cred_name = utils.uuid4()
+    name = utils.uuid4()
+    hosts = '127.0.0.1'
+    port = default_port_for_source(source_type)
+    cred_add(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+            'type': source_type,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    qpc_source_add = pexpect.spawn(
+        'qpc source add --name {} --cred {} --hosts {} --type {}'
+        .format(name, cred_name, hosts, source_type)
+    )
+    assert qpc_source_add.expect('Source "{}" was added'.format(name)) == 0
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == 0
+
+    source_show(
+        {'name': name},
+        '{{\r\n'
+        '    "credentials": \[\r\n'
+        '        {{\r\n'
+        '            "id": \\d+,\r\n'
+        '            "name": "{}"\r\n'
+        '        }}\r\n'
+        '    \],\r\n'
+        '    "hosts": \[\r\n'
+        '        "{}"\r\n'
+        '    \],\r\n'
+        '    "id": \\d+,\r\n'
+        '    "name": "{}",\r\n'
+        '    "port": {},\r\n'
+        '    "source_type": "{}"\r\n'
+        '}}\r\n'
+        .format(cred_name, hosts, name, port, source_type)
+    )
+
+    qpc_source_edit = pexpect.spawn(
+        'qpc source edit --name {} --hosts {}'
+        .format(name, new_hosts)
+    )
+    assert qpc_source_edit.expect(
+        'Source "{}" was updated'.format(name)) == 0
+    assert qpc_source_edit.expect(pexpect.EOF) == 0
+    qpc_source_edit.close()
+    assert qpc_source_edit.exitstatus == 0
+
+    if new_hosts.endswith('0/24'):
```

Comment:
Be aware that edit should not be possible for CIDR and ansible range notations https://github.com/quipucords/quipucords/issues/449. I still have to think how to handle this in this test or create a separated one for that.

Reply:
Yeah, there are a couple of cases that apply to both vcenter and satellite and that's why I broke some tests in the API side out to "network" and "host_manager" types, so I could treat vcenter and satellite together.